Description: Breaking: drop node v10/v13/v15
  * Update relative-module-resolver.js
  * Update package.json
  * Update README.md
  * Update getting-started.md
Origin: upstream, https://github.com/eslint/eslint/commit/f3cb320
Bug: https://github.com/eslint/eslint/issues/14023
Bug: https://github.com/eslint/eslint/issues/14592
Author: 薛定谔的猫 <weiran.zsd@outlook.com>
Last-Update: 2022-01-30
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/README.md
+++ b/README.md
@@ -40,7 +40,7 @@
 
 ## <a name="installation-and-usage"></a>Installation and Usage
 
-Prerequisites: [Node.js](https://nodejs.org/) (`^10.12.0`, or `>=12.0.0`) built with SSL support. (If you are using an official Node.js distribution, SSL is always built in.)
+Prerequisites: [Node.js](https://nodejs.org/) (`^12.22.0`, `^14.17.0`, or `>=16.0.0`) built with SSL support. (If you are using an official Node.js distribution, SSL is always built in.)
 
 You can install ESLint using npm:
 
--- a/docs/user-guide/getting-started.md
+++ b/docs/user-guide/getting-started.md
@@ -13,7 +13,7 @@
 
 ## Installation and Usage
 
-Prerequisites: [Node.js](https://nodejs.org/en/) (`^10.12.0`, or `>=12.0.0`) built with SSL support. (If you are using an official Node.js distribution, SSL is always built in.)
+Prerequisites: [Node.js](https://nodejs.org/en/) (`^12.22.0`, `^14.17.0`, or `>=16.0.0`) built with SSL support. (If you are using an official Node.js distribution, SSL is always built in.)
 
 You can install ESLint using npm:
 
--- a/lib/shared/relative-module-resolver.js
+++ b/lib/shared/relative-module-resolver.js
@@ -5,13 +5,7 @@
 
 "use strict";
 
-const Module = require("module");
-
-/*
- * `Module.createRequire` is added in v12.2.0. It supports URL as well.
- * We only support the case where the argument is a filepath, not a URL.
- */
-const createRequire = Module.createRequire || Module.createRequireFromPath;
+const { createRequire } = require("module");
 
 module.exports = {
 
--- a/package.json
+++ b/package.json
@@ -133,6 +133,6 @@
   ],
   "license": "MIT",
   "engines": {
-    "node": "^10.12.0 || >=12.0.0"
+    "node": "^12.22.0 || ^14.17.0 || >=16.0.0"
   }
 }
